### ISSUE TITLE
Improve email communication(s) after sign-up

### ISSUE THREAD

**Issue by thisandagain** [MozillaFoundation/plan#242]
### RACI

* Phase: Write / Build / Deploy
* Owner: @HPaulJohnson 
* Decision Maker: @HPaulJohnson 
* Design Lead: n/a
* Development Lead: @ScottDowne 
* Quality Verifier: @thisandagain @valianttry 

Our communication after sign-up should draw users back into the product and ideally come after the user has stopped engaging with Webmaker (either via a standard delay or dynamically). In addition, we need to start building up some expertise in what works / what doesn't in regards to email messaging.

Current post sign-up email:
```
Hi home61,

Welcome to Mozilla Webmaker! Thank you for joining us to explore, play and create together to build a web that's open and made by everyone.

You can get started right now with these free tools:

X-Ray Goggles allow you to see the code behind any webpage and remix elements with a single click, swapping in your own text and images.
Thimble, an easy-to-use HTML and CSS editor, lets you learn code by writing it directly in your browser to create your own webpage.
Popcorn Maker helps you combine content from across the web like GIFs, music, photos and videos into cool mashups that you can annotate and share.
I'm so glad you've joined the Webmaker community. If you have questions, feel free to email me or tweet @webmaker. I'm looking forward to seeing what you make on the web!

Cheers,
Amira
Webmaker Community Manager

If you do not wish to receive any more emails from Mozilla Webmaker you can change your email preferences.
```

### Audience
Newly registered users of webmaker.org

### Success
Primary: 
    - Improve AU to EU conversion rate
    - Make it easier for us to run these types of experiments in the future

Secondary: 
    - Learn what works / what doesn't in regards to email communication with new users

### Vision
As a baseline I'd like to see some updates to the post sign-up template (text-only / above) but ideally we could get a more robust system in-place that would allow for better tracking of open / click-through rates as well as allow for more control of timing.

**Comment by thisandagain** [comment]
One technology that I've used in the past as to make the templates more accessible as well as easy to track is http://apostle.io/

**Comment by ScottDowne** [comment]
Is there something I can start on with this?

**Comment by thisandagain** [comment]
@ScottDowne @HPaulJohnson Can you two sync-up?

**Comment by ScottDowne** [comment]
We are, but I was hoping there was something that could be dropped in the ticket vs vidyo?

**Comment by thisandagain** [comment]
So I think the big question for me is if we can replace / augment our existing transactional email system with something that allows for:
- Easy editing by the marketing & product team
- A/B testing
- Tracking of open and click rates

[Apostle](http://apostle.io/) is one such option that I'd vouch for, but we could also get some of these features out of other tools like [Mandrill](https://mandrill.com/) as well. With either option, some documentation dropped into this issue about how our current system is architected and your opinions on 3rd party options would be really helpful.

**Comment by HPaulJohnson** [comment]
@ScottDowne and I met this afternoon. Next steps: 

- Scott to investigate apostle option, talk to internal stakeholders about Sawmill capabilities and come back with recommendation for tomorrow's standup. 
- Timeline for remainder of sprint to be detailed after tomorrow's sprint.
- I will talk to @simonwex about legal/approval process for agreements with 3rd party software providers like Apostle
- New copy to be ready for review by Monday 1/26. 

Daily standup scheduled between now and end of sprint for 11:30 am PT in my vidyo.

**Comment by simonwex** [comment]
@adamlofting do you know how many emails we send per month?

@jbuck do you have an overview diagram including sawmill, webmaker-mailroom & the templates?

**Comment by simonwex** [comment]
@ScottDowne feel free to book time with me if you want to brainstorm.

**Comment by adamlofting** [comment]
As far as I know our current transactional emails don't produce any data (or if they do, I just haven't seen it). If there is any data, I'd love to look at it.

I've used Mandrill in the past and the analytics was good, but I had to build a custom editor for the content. It looks like Apostle saves us from doing that, and having A/B testing built in is excellent :metal:.

I'd like to add a requirement to this task that:
* The links in transactional emails that point back to Webmaker should include GA tracking parameters so we can better understand this channel in the scope of our overall web analytics. We can automate this programmatically but you can use this for an example of what those parameters look like: http://adamlofting.github.io/campaignurl/

**Comment by jbuck** [comment]
@simonwex Nope, no diagrams. But conceptually it'll be really easy to use apostle.io in place of webmaker-mailroom

**Comment by thisandagain** [comment]
@adamlofting @simonwex Amazon SES kicks out some basic metrics via CloudWatch. We might be able to get a sense for monthly volume from that.

**Comment by ScottDowne** [comment]
Apostle demo: https://github.com/mozilla/sawmill/pull/35/files

To test I have been doing this:

Create a file called test.js

Content of file are:
```
var worker = require("./worker");
var apostle = require('apostle.io');
apostle.domainKey = "111c1d90d76663230994fa33e64b068ae11c5302";

worker.send_new_user_email(apostle)(null, {
  event_type: "create_user",
  data: {
    email: "something.something@gmail.com",
    username: "Joe Son"
  }
}, function(err) {
  if (err) {
    console.log(err);
    return;
  }
  console.log("email sent");
});
```

Then run `node test.js`

**Comment by simonwex** [comment]
We don't have any SES stats in CloudWatch.

@jdotpz doable to figure out how many emails we send a month on average?

**Comment by jdotpz** [comment]
Sure, I'm betting our billing info will have it.  I'll look it up tonight or in the am and shoot it over.

**Comment by jdotpz** [comment]
TLDR: For the last 3 months of 2014, we averaged 1029 emails per day, or 31584 emails per month.
That said, we're already way over that in Jan, where we are averaging 1770 emails per day, or 54889 emails per month.  

The data:
Date                          # sent
2014/01/31 23:59:59 10277
2014/02/28 23:59:59 7045
2014/03/31 23:59:59 11969
2014/04/30 23:59:59 9340
2014/05/31 23:59:59 8656
2014/06/30 23:59:59 10293
2014/07/31 23:59:59 15289
2014/08/31 23:59:59 51648
2014/08/31 23:59:59 62000
2014/09/30 23:59:59 33063
2014/10/31 23:59:59 23856
2014/11/30 23:59:59 41363
2014/12/31 23:59:59 29536
2015/01/31 23:59:59 38954

**Comment by jbuck** [comment]
I suspect a large part of that is that login3 is much less friction to sign
in.

**Comment by thisandagain** [comment]
Yeah... new account sign-ups are way up post splash / login changes: https://mozillafoundation.geckoboard.com/dashboards/F62088172D822E2A

**Comment by simonwex** [comment]
### Notes from Friday Jan 23

*Action Items:*
 * [ ] Scott to verify that we, in fact currently don't send localized emails
 * [ ] Simon to contact Apostle re: localizing functionality in email
 * [ ] Paul file bug related to localizing emails in the future
 * [ ] Simon to kick off legal review process of Apostle

**Comment by HPaulJohnson** [comment]
@thisandagain, question about content of an updated email: 

In the new version of the email where should we point people in our CTA? A specific x-ray goggles activity is my preference but we can test to optimize engagement/retention. Other options - given current site functionality - would be tools, explore or profile. 

I'll work around this for now as I update copy but resolution would be good early next week.

**Comment by simonwex** [comment]
I'm moving this to P2, @ScottDowne is going to be focusing on #263 (timing-critical) next week and getting the planning work done for this as his time permits.

**Comment by ScottDowne** [comment]
@HPaulJohnson @thisandagain @simonwex I heard back from Apostle.io about privacy. There is currently no way to use the email authoring UI or AB testing without sending the user emails to a third party.

What does this mean for us?

**Comment by davidascher** [comment]
We either need to do a legal vendor review of apostle.io or write something
similar ourselves, methinks.

**Comment by simonwex** [comment]
@thisandagain introduced us to Mal, the founder. I'll email him and see if they could prioritize a UUID-based call, in conjunction we should kick-off a review. @ScottDowne let's do that "paperwork" together today.

**Comment by simonwex** [comment]
Emailed Mal and CC'd scott.

**Comment by OpenMatt** [comment]
* @HPaulJohnson : status update?

**Comment by HPaulJohnson** [comment]
This was bumped to next heartbeat as devs were re-assigned to Net Neutrality call tool project. On today's TPS call we can decide if that's far enough along to restart work.

**Comment by simonwex** [comment]
Let's chat about this during our TPS on Monday. It's likely going to come down to this or #160.

**Comment by thisandagain** [comment]
Moving #160 to the backlog.

**Comment by adamlofting** [comment]
On a related note, this new dashboard shows the number of new users p/day who opt-in to receive email communications (it's ~50% of sign-ups):

* https://metrics.webmaker.org/dashboard/email

I suspect the remainder of the daily emails shown in the stats above relate to things like login emails / reminders etc.

**Comment by thisandagain** [comment]
I propose this be a P2 for this sprint given the technical debt reduction and parallel login / sign-up work.

**Comment by OpenMatt** [comment]
* still need to pick a vendor, iron out final complexity / questions. 
* could roll this into log-in work this heartbeat
* also need to consider teach.webmaker.org email requirements

**Comment by OpenMatt** [comment]
(will leave this as an open planning ticket for now -- @HPaulJohnson please close and file as needed once the kick-off happens / next steps get logged, etc.)

**Comment by HPaulJohnson** [comment]
Summary for easy reading and as we explore whether this ticket can be rolled into #53. 

We need a better solution to support transactional/triggered emails from webmaker.org and teach site. The welcome email triggered at login is the first example of this but there can/will be many more instances. 

The key requirements are: 

- Content management system that allows marketing/engagement/comms to add, modify or remove email content without engineering support.
- Ability to do A/B testing within these automated emails
- Ability to track and measure performance (deliverabiility, open, click, unsub, etc).

In the first go around with this ticket we looked at a small number of 3rd party solutions, giving the most attention to Apostle.io. Apostle was tested by Scott and Simon had begun exploring legal/approval process for agreement before this ticket was reprioritized in favor of Net Neutrality work that was more time sensitive. 

@secretrobotron to decide whether we close this ticket and move to #53 or keep this part here.

**Comment by secretrobotron** [comment]
:+1: 
Confirmed with @ScottDowne and @simonwex:
* Apostle is the way to go
* Keep it separate from teach login integration for this heartbeat (and until review is done)
* Start sec review on Apostle right away ( @simonwex )
* @ScottDowne's [pull request](https://github.com/mozilla/sawmill/pull/35/files) is a draft implementation

**Comment by secretrobotron** [comment]
@simonwex any news on Apostle review?

**Comment by ScottDowne** [comment]
@secretrobotron I'm submitting it. I had some questions about @simonwex 's instructions so once I get a few answers I should be good to kick this off. The weekend caused a delay here.

I need to know when we expect this to release. I feel like March 27th (listed in this bug is too soon for the review?) But not sure how long I should expect a review to take.

**Comment by secretrobotron** [comment]
This ticket will definitely have a follow-up. March 27 isn't an implementation deadline. Let's just make sure the process is started this week.

**Comment by ScottDowne** [comment]
The process to start is a form that has a due date as a required field.

I need to put *something* in to start the process.

**Comment by secretrobotron** [comment]
Ah. @HPaulJohnson or @hannahkane might have an opinion. Would be nice to have the review done by March 27, but I'm not sure how long they might take. @simonwex might have some idea though.

**Comment by davidascher** [comment]
Just say ASAP / March 27
​

**Comment by OpenMatt** [comment]
status?

**Comment by OpenMatt** [comment]
@ScottDowne is doing vendor review. He will update this ticket this morning.

**Comment by ScottDowne** [comment]
@OpenMatt nothing has happened in the security review bug, but the legal review bug has some questions for us/apostle that I need to see how many I can answer and how many I need to push to other people.

(1) What do we intend to use these emails for?
(2) They suggested ExactTarget as an alternative to Apostle.
(3) Linked us this: https://www.mozilla.org/en-US/privacy/websites/ calling out the ability to let users view text only emails.
(4) What analytics data do they receive from emails?
(5) What data to the collect and store, other than email address?
(6) In what country is this data stored? For how long is it stored?
(7) Can it be deleted at the user's request? Or Mozilla's request?

**Comment by simonwex** [comment]
Hey @ScottDowne, 1) is covered above, 2) we should talk about and the rest you should get into contact Mal about.

On ExactTarget v.s. Apostle, I believe this is worth a discussion. It is a salesforce company, so integrations with our CRM is likely. MoCo is already making use of it, as are we in some cases.

@thisandagain, I think ExactTarget would suit this use-case, do you agree?

@davidascher do you know how we can figure out what pricing would be for ExactTarget for us?

**Comment by valianttry** [comment]
Cross posting this response:
The ExactTarget agreement with Moco is that we pay by the # of emails mofo sends using ExactTarget, and we've already banked plenty of email in our estimates (including Webmaker messaging and other projects like Open News emails; etc). Overage kicks in if we've under-estimated. I don't know how much the Basket customization would apply to our implementation since Basket was built at a time before Salesforce bought ExactTarget and began a deeper integration across both. We're implementing Salesforce along with ExactTarget, so the integration might be simpler (I don't know for sure). 

I *strongly* urge us to implement ExactTarget for all Mofo's email marketing -- I thought that was the intention (but clearly there's a disconnect somewhere). Since ExactTarget is already through legal and security review, we already have a signed contract and can begin implementation -- and based on my understanding of Webmaker's transactional email strategy --  it should meet the needs.

**Comment by valianttry** [comment]
Here's a blog post about ExactTarget, Journey Builder, and Transaction Builder (all part of the Exact Target suite): 
http://blogs.perficient.com/digitaltransformation/2013/11/18/dreamforce-13-email-marketing-with-exacttarget-marketing-cloud/

"Transaction Builder is used when you wanted to send triggered and transactional emails.  For example, when a customer interacts with you, you want to send them emails based on actions they take.  For example, when you rent a movie at Redbox, you receive an email as soon as you get the movie from the machine.  ExactTarget sends millions of these transactional emails for Redbox.
Triggered and Transactional emails are the most opened emails.
ExactTarget uses a robust API to allow your internal systems to connect to the transaction builder.
...


    This includes dynamic content, testing & optimization, and mobile content for email and mobile messages.
    The content system is built on an open platform that allows you to create “gears” that can be included in the mail.
    Nike used ExactTarget to go from 2 customized emails (men vs women) to over 1.3 million versions without additional staff.
    Content can be rendered at send time or at run time. Run time content can be customized for the user every time the email is opened.
"

**Comment by thisandagain** [comment]
It sounds like this may cover our use case. I'd love to get into the details as I'm having trouble finding documentation but if we can use ExactTarget I'd much prefer to have our transactional and marketing emails together within the same system. Thanks for the catch @valianttry !

Will schedule a follow-up so we can dig in.

**Comment by valianttry** [comment]
@thisandagain sounds brilliant -- also I want to talk to you about localization management... want me to schedule 30m for you and I for these 2 things? Who else should join us?

**Comment by HPaulJohnson** [comment]
I would like to join!

**Comment by thisandagain** [comment]
:+1:

**Comment by ScottDowne** [comment]
Another question. I don't think my answer was enough. It's about Apostle still 

"Can you provide more details about the analytics data and the collection that entails? What specific data is collected?  What fields are collected for the analytics and AB testing? Are there unique identifiers?"

Is there any more to this other than we want to run AB tests and track clicks on links in the email?

**Comment by OpenMatt** [comment]
* Thanks for reaching out on this Andrea
* Sounds like next step is: Andrew, Andrea, Paul (+ others?) will meet today / tmrw. To compare requirements and capabilities for ExactTarget vs. Apostle. Then file the results here. 
* Given the overlap and potential for co-ordination, I've suggested adding Andrea as a second quality-verifier to this ticket. To ensure what we pick here squares with transactional email strategy in other parts of the org.

**Comment by OpenMatt** [comment]
* One question this raises: should Scott continue with vendor review on Apostle in the mean time? Or wait for the results of that meeting?

**Comment by ScottDowne** [comment]
@OpenMatt I think we should wait before I continue much.

A question I have is should we notify people in the vendor review bug that we're considering exactTarget?

**Comment by thisandagain** [comment]
Looked at ExactTarget with Andrea today. Waiting on access to the sandbox to answer a few final questions but yes, I recommend that put the Apostle vendor review on-hold.

**Comment by ScottDowne** [comment]
@thisandagain I think that's the right decision.

I can update the vendor review ticket tomorrow saying we're on hold.

**Comment by thisandagain** [comment]
I got access to the sandbox and went through the documentation today. My recommendation is that we use ExactTarget (specifically, the Journey Builder feature) rather than Apostle to improve and monitor our transactional emails:

#### Pros
- Same system will eventually be used to track all of our outbound email which will greatly improve the visibility / consistency of our communications with users
- Has already been through vendor review / can start implementation now
- More robust analytics tools
- More robust targeting options
- Supports localization
- Supports SMS and Push Notifications
- Gives us a head start on getting acquainted with Salesforce ahead of the EOY campaign

#### Cons
- *Incredibly* more complex with a *much* higher learning curve
- Takes longer to configure due to increased complexity
- Editing individual email templates is slightly more complex / tedious but still does not require writing code

**Comment by thisandagain** [comment]
Huge "thank you" to @valianttry for catching this ticket and helping us get familiar with Salesforce! :smile: :cat: :cake: :boom:

**Comment by valianttry** [comment]
@thisandagain  I'm SO glad I could be helpful and (hopefully) make life easier for all of you. Note that it is very possible/probably that I'll be hiring someone on the digital team who can help setup emails and provide support for ExactTarget (possibly the fundraising associate that's currently posted). 

:clap: :grinning:

**Comment by thisandagain** [comment]
Update: Working with the team at MOCO to setup a training program which is really the first step before we can get these transactional emails moved. @adamlofting is taking the lead on developing our plan for Salesforce across the org.

**Comment by adamlofting** [comment]
This ticket has a lot of history and discussion that we can archive now.  :package:

I've filed a new ticket to track the task of implementing this: https://github.com/MozillaFoundation/plan/issues/433

If your name isn't currently against that new ticket, but you want to follow along, please comment on that new ticket so you get the notifications (or "transaction emails" as it were :wink: ).
